Use NSOperation and NSOperationQueue to download image tiles from the web in concurent threads.

vyskocil opened the pull request:
The use of NSOperation/NSOperationQueue to download image tiles from web helps to unload work from the main thread and permit a smoother map scrolling, particularly in iOS5.

vyskocil committed:
Use NSOperation and NSOperationQueue to download image tiles from the web in concurent threads.

incanus commented:
The new `RMURLConnectionOperation` class isn't included in the Xcode project...

vyskocil committed:
- retain/release NSDate lastUsedTime before passing it to dao
- protect images set enumeration with @synchronized(self)
- properly release NSURLRequest

vyskocil committed:
Add RMURLConnectionOperation.m and RMURLConnectionOperation.h to the project

michaeltyson commented:
Nice one! One note: I had to make some adjustments to make sure it performs certain NSURLConnection delegate methods on the main thread, otherwise it causes wacky problems.
